compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -375,6 +375,20 @@ impl<'tcx> Body<'tcx> {
         (&mut self.basic_blocks, &mut self.local_decls)
     }
 
+    #[inline]
+    pub fn basic_blocks_and_source_scopes_mut(
```

Comment:
I wasn't particularly happy to introduce yet another variant of `basic_blocks_and_...`, so I opened #98930 that would remove a need for it.